### ISSUE TITLE
Update interface for superresolution

### ISSUE THREAD

**Issue by alam-shahul** [alam-shahul/popari#14]
It is currently unclear how best to perform the superresolution aspect of the Popari algorithm. Some pitfalls include:

- Some hyperparameters are not fully exposed on the Popari model object
- It is unclear whether multiple rounds of superresolution with different learning rates are required for convergence; this should be profiled

**Comment by alam-shahul** [comment]
With the addition of the `Trainer` object (see #25), this can all be added here. Most probably, we can use some sort of learning rate schedule with many iterations to resolve the issue of choosing a learning rate. From experience, it seems like an initial learning rate of `1e-1`  and `10000` epochs is a good starting point; the only issue is the memory leak (#16), which makes it hard to increase the number of epochs to an amount sufficient for all use cases.
